src/main/java/edu/ucla/library/prl/harvester/JobResult.java

```diff
@@ -63,13 +76,14 @@ public JobResult(final int aJobID, final OffsetDateTime aStartTime, final int aR
      * @param aJsonObject A job result represented as JSON
      * @throws InvalidJobResultJsonException If the JSON representation is invalid
      */
-    @SuppressWarnings({ "PMD.AvoidLiteralsInIfCondition", "PMD.CyclomaticComplexity" })
+    @SuppressWarnings({ "PMD.AvoidLiteralsInIfCondition", "PMD.CognitiveComplexity", "PMD.CyclomaticComplexity" })
```

Comment:
Wondering if you feel these are tech debt or whether the thresholds should be increased?

Reply:
I think these constructor implementations must basically be as they are in order to do the validation steps required to make illegal instances impossible to create. They don't particularly bother me, because the shapes/patterns of the independent validation steps are so similar (i.e., an `if` block with either an `if/else` or a `try/catch` (with the `else` or `catch` throwing), followed by an `else` block that throws), but I could see there being private helper methods to validate each parameter.

My opinion on these PMD complexity warnings is that while they are useful for pointing out code that might best be rewritten, we shouldn't feel compelled to let them dictate how we write code, but rather: consider ourselves, the human readers, as the ultimate authority on what to do about them, on a case-by-case basis (assuming that they're used "infrequently enough", which I think they are; I count four instances each of CognitiveComplexity and CyclomaticComplexity, three of them overlapping, so five methods/constructors total). I suspect we may disagree on this.

So, I (myself) feel neither that these represent tech debt nor that the threshold should be increased, since I think that the default threshold is _generally_ appropriate elsewhere in the codebase (since for the most part, it isn't being hit); just that in these few instances, I think it's appropriate to ignore the warnings. But it is possible that having the opportunity of "first code" has clouded my judgment here, and that you might suggest (for some reason that might not be apparent to me) that they be written differently.